src/app/rosetta/lib/construction.ml

```diff
@@ -1071,19 +1129,39 @@ module Submit = struct
               ()
             with
             | `Successful res ->
+               [%log info] "Gql payment succeful";
                let cache = Lazy.force submitted_cache in
                Cache.add cache txn;
                let (`UserCommand payment) = res#sendPayment#payment in
                M.return payment#hash
             | `Failed e ->
+               [%log info] "Gql payment failed";
                let cache = Lazy.force submitted_cache in
                if
                  ([%equal: Errors.Variant.t] (Errors.kind e) `Transaction_submit_bad_nonce) &&
                    Cache.mem cache txn
                then
                  M.fail (Errors.create `Transaction_submit_duplicate)
                else
-                 M.fail e )
+                 (
+                 let cmd = txn.command in
+                 [%log info] "trying a signed command";
+                 match%bind
+                   env.db_transaction_exists
+                     ~nonce:txn.nonce
+                     ~source:(let (`Pk s) = cmd.source in s)
+                     ~receiver:(let (`Pk r) = cmd.receiver in r)
+                     ~amount:
+                        (Option.value
+                          ~default:Unsigned_extended.UInt64.zero cmd.amount)
+                     ~fee:cmd.fee
+                 with
+                 | true ->
+                     [%log info] "it exists";
```

Comment:
Are these logs for debugging? Will they remain in the merged code?

Reply:
shoot, thanks 👍 . I always forget about the logs